### PR TITLE
unicode-rs: initial integration.

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Performs an umbrella integration for Unicode-related repositories:
- [Unicode-normalisation](https://crates.io/crates/unicode-normalization) (23 million downloads)
- [Unicode-segmentation](https://crates.io/crates/unicode-segmentation) (17 million downloads)

Unicode normalisation uses unsafe https://github.com/unicode-rs/unicode-normalization/search?q=unsafe code whereas unicode-segmentation does not.

@Manishearth I have set up continuous fuzzing of unicode-rs repos. The benefits are that the fuzzer will be run on a regular basis. Is this something you would be happy to have set up? If so, the only thing needed from your end is an email that will receive the bugs reports - notice this email should be attached to a Google account for you to see the reports. I can also move the fuzzers for `unicode-segmentation` to the upstream repo.

### CODE REVIEW

File: projects/unicode-rs/unicode-segmentation-fuzz/fuzz_targets/fuzz_target_1.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use unicode_segmentation::UnicodeSegmentation;
+
+fuzz_target!(|data: &[u8]| {
+    if let Ok(s) = std::str::from_utf8(data) {
+        let _g = s.graphemes(true).collect::<Vec<&str>>();
```
Comment (Manishearth) [inline projects/unicode-rs/unicode-segmentation-fuzz/fuzz_targets/fuzz_target_1.rs:7]: for this and split_word_bounds perhaps we can assert that it is the same as the original string?

**Review by Manishearth — APPROVED** [review]
Looks good, please use manishsmail@gmail.com.

Happy to upstream fuzzers as well; whatever works for y'all!

File: projects/unicode-rs/unicode-segmentation-fuzz/fuzz_targets/fuzz_target_1.rs (hunk 1 above)
Comment (DavidKorczynski) [reply projects/unicode-rs/unicode-segmentation-fuzz/fuzz_targets/fuzz_target_1.rs]: Sounds good to me! We have the fuzzers upstream now so let's do the changes there!

**Comment by DavidKorczynski** [comment]
@inferno-chromium @oliverchang @jonathanmetzman calling on the current sheriff this one is good to go.

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**